probe-rs/src/debug/unit_info.rs

```diff
@@ -142,43 +134,48 @@ impl UnitInfo {
                 continue;
             }
 
-            let mut ranges = debug_info.dwarf.die_ranges(&self.unit, current)?;
+            let mut gimli_ranges = debug_info.dwarf.die_ranges(&self.unit, current)?;
+            let mut die_ranges = Vec::new();
+            while let Ok(Some(gimli_range)) = gimli_ranges.next() {
+                die_ranges.push(gimli_range.begin..gimli_range.end);
+            }
 
-            while let Ok(Some(range)) = ranges.next() {
-                if !range.contains(address) {
-                    continue;
-                }
-                // Check if we are actually in an inlined function
+            if !die_ranges.iter().any(|range| range.contains(&address)) {
+                continue;
+            }
 
-                // We don't have to search further up in the tree, if there are multiple inlined functions,
-                // they will be children of the current function.
-                abort_depth = current_depth;
+            // Check if we are actually in an inlined function
 
-                // Find the abstract definition
-                let Ok(Some(abstract_origin)) = current.attr(gimli::DW_AT_abstract_origin) else {
-                    tracing::warn!("No abstract origin for inlined function, skipping.");
-                    return Ok(vec![]);
-                };
-                let abstract_origin_value = abstract_origin.value();
-                let gimli::AttributeValue::UnitRef(unit_ref) = abstract_origin_value else {
-                    tracing::warn!(
-                        "Unsupported DW_AT_abstract_origin value: {:?}",
-                        abstract_origin_value
-                    );
-                    continue;
-                };
+            // We don't have to search further up in the tree, if there are multiple inlined functions,
+            // they will be children of the current function.
+            abort_depth = current_depth;
```

Comment:
I find this depth logic to be a very confusing version of `if depth  < 0`. Not sure what the intent was.

Reply:
I'm guessing one of the upper continues make this more than an if, but I'm not at all sure what's happening or why.

Reply:
I didn't write this code, but am pretty confident that it results from how the gimli DIE's are traversed with the `.dfs()` iterator.